pkg/pubsub/pubsub_client.go

```diff
@@ -0,0 +1,119 @@
+package pubsub
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/pubsub/dapr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func init() {
+	flag.Var(pubSubs, "pub-sub-tool", "Preferred pub-sub tool, e.g dapr, rabbitmq. This flag can be declared more than once. Omitting will default to not using any pub sub tool.")
+}
+
+var pubSubs = newPubSubSet(map[string]PubSub{
+	dapr.Name: &dapr.Dapr{},
+},
+)
+
+// PubSub is the interface that wraps pubsub methods.
+type PubSub interface {
+	// Publish single message over a specific topic/channel
+	Publish(data interface{}, topic string) error
+
+	// Publish a batch of messages over a specific topic/channel
+	PublishBatch(data interface{}, topic string) error
+
+	// Initiate the pub sub client
+	NewClient(ctx context.Context, client client.Client) error
+
+	// Get the name of pub sub tool
+	GetName() string
+
+	// Determines if publish batching is enabled or not
```

Comment:
IsBatchingEnabled should probably not be exposed as a user-facing interface, unless this is meant to be the interface for the driver?

Reply:
Yeah, it is intended for the interface driver. I think it has additional flexibility when it comes to custom drivers as one can use this function to return `true`/`false` always and only worry about the correct approach for them (i.e either `Publish` or `PublishBatch` for all their messages)